src/components/theme/View/View.jsx

```diff
@@ -203,8 +203,17 @@ class View extends Component {
    */
   render() {
     const { views } = config;
-    if (this.props.error && this.props.error.code === 301) {
-      const redirect = flattenToAppURL(this.props.error.url).split('?')[0];
+    const status = __SERVER__
+      ? this.props.error?.status
+      : this.props.error?.code;
+    // Checking to see if it's a 3XX HTTP status code. error.status only works on the server
+    if (status?.toString()?.[0] === '3') {
+      const redirectUrl = __SERVER__
+        ? this.props.error.response.headers.location
+        : this.props.error.url;
+      let redirect = flattenToAppURL(redirectUrl);
+      // We can hit situations where we end up being redirected to an api route. We don't want that so lets remove ++api++.
+      redirect = redirect.replace('/++api++', '');
       return <Redirect to={`${redirect}${this.props.location.search}`} />;
```

Comment:
you will need to set a redirect code which is the same the one you got from the api call since later restapi will give different codes depending if you are logged in or not

Reply:
Yep, this will be done in a later PR as mentioned in the description